### PR TITLE
feat: Add option to omit anonymous users from index and identify events

### PR DESCRIPTION
**Pull request by keelerm84**

### CODE REVIEW

File: contract-tests/Cargo.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ futures = "0.3.12"
 hyper = { version = "0.14.19", features = ["client"] }
 hyper-rustls = { version = "0.24.1" , optional = true, features = ["http2"]}
 hyper-tls = { version = "0.5.0", optional = true }
+launchdarkly-server-sdk-evaluation = { git = "https://github.com/launchdarkly/rust-server-sdk-evaluation", branch = "mk/sc-249036/omit-anon" }
```
Comment (keelerm84) [inline contract-tests/Cargo.toml:20]: Temporary until the related changes are published.

File: launchdarkly-server-sdk/Cargo.toml [hunk 2]
```diff
@@ -25,7 +25,7 @@ lazy_static = "1.4.0"
 log = "0.4.14"
 lru = { version = "0.12.0", default-features = false }
 ring = "0.17.5"
-launchdarkly-server-sdk-evaluation = "1.2.0"
+launchdarkly-server-sdk-evaluation = { git = "https://github.com/launchdarkly/rust-server-sdk-evaluation", branch = "mk/sc-249036/omit-anon" }
```
Comment (keelerm84) [inline launchdarkly-server-sdk/Cargo.toml:28]: Same here of course.

File: launchdarkly-server-sdk/src/events/dispatcher.rs [hunk 3]
```diff
@@ -219,30 +219,56 @@ impl EventDispatcher {
                     self.outbox.add_event(OutputEvent::FeatureRequest(inlined));
                 }
             }
-            InputEvent::Identify(identify) => {
+            InputEvent::Identify(mut identify) => {
+                if self.events_configuration.omit_anonymous_contexts {
+                    match identify.base.context.without_anonymous_contexts() {
+                        Ok(context) => identify.base.context = context,
+                        Err(e) => {
+                            error!("Failed to omit anonymous contexts: {}", e);
```
Comment (kinyoklion) [inline launchdarkly-server-sdk/src/events/dispatcher.rs:227]: Should we really have an error here? This seems like an expected condition.

Comment (keelerm84) [reply launchdarkly-server-sdk/src/events/dispatcher.rs]: Fair point. I have removed it.

**Review by kinyoklion — APPROVED** [review]

**Pull request merged**